src/coreclr/pal/inc/pal.h

```diff
@@ -3618,6 +3618,22 @@ Define_InterlockMethod(
     ((PVOID)(UINT_PTR)InterlockedCompareExchange((PLONG)(UINT_PTR)(Destination), (LONG)(UINT_PTR)(ExChange), (LONG)(UINT_PTR)(Comperand)))
 #endif
 
+#if defined(HOST_64BIT) && defined(FEATURE_CACHED_INTERFACE_DISPATCH)
+FORCEINLINE uint8_t _InterlockedCompareExchange128(int64_t volatile *pDst, int64_t iValueHigh, int64_t iValueLow, int64_t *pComparandAndResult)
```

Comment:
This can also be part of ``src/coreclr/minipal`, and we can remove the copy nativeaot.

nit: we added more comments in the latest version of nativeaot copy, so please resync.

Reply:
`src\native\minipal` may be more appropriate.

Reply:
@jkotas unfortunately, having support for _InterlockedCompareExchange128 is not something that all of our 64bit platforms ever will have. In particular, we can't really support it for Linux X64 as a number of our supported distros don't intend to bump up their minimum version any time soon, and Loongarch would require us to bump up the minimum CPU version. So my intention is to ALWAYS keep this under a FEATURE_CACHED_INTERFACE_DISPATCH #ifdef, which would make me uncomfortable with pushing the header down into the native minipal. I then intend (before I actually merge this logic) to make it so that the support for Cached Interface dispatch is only enabled for chk/debug builds in our current build world.

Reply:
Ok, sounds good.

> Loongarch would require us to bump up the minimum CPU version. 

Risvc has this problem too.

Reply:
Yes, I see that the comment that @am11 wanted me to copy around discusses the RISC-V situation. Regardless, I'm going to consider this issue resolved, as the plan doesn't currently involve actually using this logic on a problematic OS/architecture pair.